test/integration/associations/dont-modify-options.test.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+/* jshint -W030 */
+var chai = require('chai')
+  , expect = chai.expect
+  , Support = require(__dirname + '/../support')
+  , DataTypes = require(__dirname + '/../../../lib/data-types')
+  , Sequelize = require('../../../index');
+
+describe(Support.getTestDialectTeaser('associations'), function() {
+  describe('Test options.foreignKey', function() {
+    beforeEach(function() {
+
+      this.A = this.sequelize.define('A', {
+        id: {
+          type: DataTypes.CHAR(20),
+          primaryKey: true
+        }
+      });
+      this.B = this.sequelize.define('B', {
+        id: {
+          type: Sequelize.CHAR(20),
+          primaryKey: true
+        }
+      });
+      this.C = this.sequelize.define('C', {});
+    });
+
+    it('should not be overwritten for belongsTo', function(){
+      var self = this;
+      return this.sequelize.sync({force: true}).then(function() {
```

Comment:
There's no need to call  `.sync` here - which means the test can be converted to a unit test